fv3core/utils/grid.py

```diff
@@ -526,8 +550,8 @@ def domain_full(self, add: Index3D = (0, 0, 0)):
         Returns the shape of the full domain including halos, plus an optional offset.
         """
         return (
-            self.ied - self.isd + add[0],
-            self.jed - self.jsd + add[1],
+            self.ied + 1 - self.isd + add[0],
+            self.jed + 1 - self.jsd + add[1],
             self.ke - self.ks + add[2],
```

Comment:
i and j get a + 1 but not the k dimension? I think ke should also be the index of the last point in the vertical dimension to be consistent with the other dimensions... the spec.grid.npz was not an index, but the number of vertical points, but if shifting to using ke to be parallel to ied and jed, it should also be an index.

Reply:
Resolved by deleting `ks` and `ke` from the object, we can use `origin[2]` and `domain[2]` instead.